VideoLocker/src/main/java/org/edx/mobile/http/Api.java

```diff
@@ -1266,4 +1270,52 @@ public Boolean enrollInACourse(String courseId, boolean email_opt_in) throws Exc
         return false;
     }
 
+
+    /**
+     * Returns course structure object from the given URL.
+     * @param courseId
+     * @param preferCache
+     * @return
+     * @throws Exception
+     */
+    public String getCourseStructure(String courseId, boolean preferCache) throws Exception {
```

Comment:
@aleffert  I will create a new PR to make that code change.  
add a service layer,  and clean up the API code.    as we will use retrofit in the near future,  API.java code is expected to be upgraded.